cosmos_rl/dispatcher/data/packer/base.py

```diff
@@ -228,3 +194,47 @@ def extend_conversation(
         """
         # By default, we always add response as assistant message
         return add_assistant_message(conversation, "" if responses else responses[0])
+
+    def save_state(self, dest_path: str) -> None:
+        pass
+
+
+class ChatDataPacker(DataPacker, ABC):
```

Comment:
For this change, the base DataPacker class used in the folder` tools/dataset` should be also changed?

Reply:
If an existing custom datapacker (that is not in cosmos-rl repo) is defined as inheriting from `DataPacker` , this change will break them?

Reply:
Thanks for the comments. To make the changes backward compatible, I'm removing `ChatDataPacker` and instead use the same `DataPacker` class in for the all of the `DataPacker` implementations in the repo.

To account for the change in the `setup` signature, I'm following the same runtime inspection as `Dataset` to call the `setup` function. The `setup` function already takes in `*args`, so removing the `tokenizer` from the signature in the `DataPacker` class is fine.

I'm also adding a new `BaseDataPacker` class which is a parent of `DataPacker` - the `BaseDataPacker` is meant to be tokenizer-agnostic.

Reply:
That is great!